scripts/send-metrics.py

```diff
@@ -91,7 +91,7 @@ def send_metrics_to_segment(event_properties, config):
 
     May throw.
     """
-    event_properties = dict(**event_properties)
+    event_properties = dict(is_test=test_mode or 'no', **event_properties)
```

Comment:
[non-blocking] 
1. The code is probably fine as-is, but I'm not sure whether people might set `test_mode` incorrectly. 
2. I find the `dict()` call takes my brain a little longer to process then just setting the key, unless I am missing something.

The following is one possible update, that could be on 2 lines or 1.
```suggestion
    is_test = 'no' if not test_mode or test_mode == 'false' else test_mode
    event_properties['is_test'] = is_test
```

Reply:
I'm not too worried about `test_mode` since the only time it should be set is in CI tests or when a dev is running the tests manually, and the tests specifically ask you to set `DEVSTACK_METRICS_TESTING=debug` if you haven't set it.

On the dict, I still haven't figured out a great way to express "add these two dictionaries together" in Python. I come from a functional programming background where I could normally just do something in the spirit of `event_properties += {"is_test": ...}`, but apparently there's no natural facility for that in Python. (What I've written is close, but will throw if the key already exists.)

The reason I haven't used `event_properties['is_test'] = ...` is that that would entail mutating the input, which *currently* would have no effect on the running of the program but is the kind of "spooky action at a distance" I try very hard to avoid if I can help it, because it makes reasoning about the code so much harder.

What would you think of this?

```python
event_properties = event_properties.copy()  # shallow copy of input
event_properties['is_test'] = test_mode or 'no'  # mark as non-test by default
```

Reply:
I now see why you had the original code. I like this latest suggestion, but am ok with whatever you like best.

Reply:
OK, I'll go with that, and try to make sure it conveys intent better than the original code did. :-)